server/middleware/auth.go

```diff
@@ -97,14 +97,33 @@ func BypassAuthCaches(ctx context.Context) bool {
 
 func GetJWTValidators(config *config.Config) []*validator.Validator {
 	jwtValidators := make([]*validator.Validator, len(config.Auth.Issuers))
-	for i, issuer := range config.Auth.Issuers {
-		log.Info().Msg(issuer)
-		issuerURL, _ := url.Parse(issuer)
-		provider := jwks.NewCachingProvider(issuerURL, config.Auth.JWKSCacheTimeout)
+	for i, issuerCfg := range config.Auth.Issuers {
+		log.Info().Msg(issuerCfg.Issuer)
+		issuerURL, _ := url.Parse(issuerCfg.Issuer)
+		var keyFunc func(ctx context.Context) (interface{}, error)
+		switch issuerCfg.Algorithm {
+		case validator.RS256:
+			{
+				provider := jwks.NewCachingProvider(issuerURL, config.Auth.JWKSCacheTimeout)
+				keyFunc = provider.KeyFunc
+				break
+			}
+		case validator.HS256:
+			{
+				keyFunc = func(ctx context.Context) (interface{}, error) {
+					return []byte(issuerCfg.SharedSecret), nil
+				}
+				break
+			}
+		default:
+			{
+				panic("Unsupported Token signature algorithm")
```

Comment:
do we need to panic? Or return an error from here?

Reply:
This is the code path that will get run during startup and not per request - so panic here is fine.

Reply:
Okay, I thought it is in middleware and will be execute on per request.